integration-tests/digitalocean/src/main/java/org/apache/camel/quarkus/component/digitalocean/it/DigitaloceanResource.java

```diff
@@ -184,35 +194,362 @@ private Action doAction(int id, DigitalOceanOperations operation) {
+        }
+        return new Action();
+    }
+
+    @DELETE
+    @Path("floatingIP/{id}")
+    public Delete deleteFloatingIp(@PathParam("id") String id) {
+        LOG.infof("delete floating IP %s", id);
+        Map<String, Object> headers = new HashMap<>();
+        headers.put(DigitalOceanHeaders.OPERATION, DigitalOceanOperations.delete);
+        headers.put(DigitalOceanHeaders.FLOATING_IP_ADDRESS, id);
+        // if pending task on droplet : API returns an exception. There should be a retry later
+        try {
+            return producerTemplate.requestBodyAndHeaders("direct:floatingIPs", null, headers, Delete.class);
+        } catch (Exception e) {
+            LOG.errorf("Enable to delete Floating IP - Please retry - error message : %s" + e.getMessage());
+        }
+        return new Delete(false);
+
+    }
+
+    @GET
+    @Path("floatingIP/actions/{id}")
+    public List<Action> getFloatingIpsActions(@PathParam("id") String id) {
+        LOG.infof("get all floating IPs actions %s", id);
+        Map<String, Object> headers = new HashMap<>();
+        headers.put(DigitalOceanHeaders.OPERATION, DigitalOceanOperations.listActions);
+        headers.put(DigitalOceanHeaders.FLOATING_IP_ADDRESS, id);
+        return (List<Action>) producerTemplate.requestBodyAndHeaders("direct:floatingIPs", null, headers, List.class);
+    }
+
+    @PUT
+    @Path("blockStorages")
+    public String createVolume(String name) {
+        LOG.infof("create volume for region %s with size %s named %s", REGION_FRANKFURT, VOLUME_SIZE_5_GB, name);
+        Map<String, Object> headers = new HashMap<>();
+        headers.put(DigitalOceanHeaders.OPERATION, DigitalOceanOperations.create);
+        headers.put(DigitalOceanHeaders.REGION, REGION_FRANKFURT);
+        headers.put(DigitalOceanHeaders.VOLUME_SIZE_GIGABYTES, VOLUME_SIZE_5_GB);
+        headers.put(DigitalOceanHeaders.NAME, name);
+        headers.put(DigitalOceanHeaders.DESCRIPTION, "volume_" + name);
+        Volume volume = producerTemplate.requestBodyAndHeaders("direct:blockStorages", null, headers, Volume.class);
+        return volume.getId();
+    }
+
+    @GET
+    @Path("blockStorages/{id}")
+    public Volume getVolumeById(@PathParam("id") String volumeId) {
+        LOG.infof("getting volume %s", volumeId);
+        Map<String, Object> headers = new HashMap<>();
+        headers.put(DigitalOceanHeaders.OPERATION, DigitalOceanOperations.get);
+        headers.put(DigitalOceanHeaders.ID, volumeId);
+        headers.put(DigitalOceanHeaders.REGION, REGION_FRANKFURT);
+        return producerTemplate.requestBodyAndHeaders("direct:blockStorages", null, headers, Volume.class);
+    }
+
+    @POST
+    @Path("blockStorages/attach/{name}")
+    public Action attachVolumeToDroplet(@PathParam("name") String volumeName, String dropletId) {
+        LOG.infof("attache volume %s to droplet %s in region %s", volumeName, dropletId, REGION_FRANKFURT);
```

Comment:
Minor typo 'attache'

Reply:
@aldettinger I've just pushed the fix typo. the only thing that I've changed